File: swift/Sources/SignalClient/zkgroup/UuidCiphertext.swift
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation
import SignalFfi

public class UuidCiphertext: ByteArray {
  public required init(contents: [UInt8]) throws {
    try super.init(contents, checkValid: signal_uuid_ciphertext_check_valid_contents)
  }
}

File: java/java/src/main/java/org/signal/zkgroup/groups/UuidCiphertext.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.groups;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class UuidCiphertext extends ByteArray {

  public static final int SIZE = 65;

  public UuidCiphertext(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
    Native.UuidCiphertext_CheckValidContents(contents);
  }

  public byte[] serialize() {
    return contents.clone();
  }

}
